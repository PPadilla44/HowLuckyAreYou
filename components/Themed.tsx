import { Text as DefaultText, View as DefaultView, ScrollViewProps as dScrollViewProps } from 'react-native';
import { Icon as DefaultIcon, IconProps as DefaultIconProps, Input as DefaultInput, InputProps as DefaultInputProps, ButtonGroup as DButtonGroup, ButtonGroupProps as DButtonGroupProps } from 'react-native-elements';
import { ScrollView as DefaultScrollView, NativeViewGestureHandlerProps, TouchableOpacity as DefaultTouchableOpacity } from "react-native-gesture-handler";

import Colors from '../constants/Colors';

import useColorScheme from '../hooks/useColorScheme';

export const useThemeColor = (
    props: { light?: string; dark?: string },
    colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) => {
    // const theme = useColorScheme();
    const theme = "dark";

    const colorFromProps = props[theme];

    return { colorFromProps: colorFromProps ? colorFromProps : Colors[theme][colorName], theme }
}

type ThemeProps = {
    lightColor?: string;
    darkColor?: string;
};

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];
export type TouchableOpacityProps = ThemeProps & DefaultTouchableOpacity['props'];
export type IconProps = ThemeProps & DefaultIconProps;
export type InputProps = ThemeProps & DefaultInputProps;
export type ScrollViewProps = ThemeProps & NativeViewGestureHandlerProps & dScrollViewProps;
export type ButtonGroupProps = ThemeProps & DButtonGroupProps;

export const Text = (props: TextProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;

    return <DefaultText style={[{ color, fontFamily: "Futura" }, style]} {...otherProps} />
}

export const View = (props: ViewProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    const shadowColor = useThemeColor({  }, "text").colorFromProps;
    return <DefaultView style={[{ backgroundColor, shadowColor }, style]}  {...otherProps} />
}

export const TouchableOpacity = (props: TouchableOpacityProps) => {
    const { style, containerStyle, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "button").colorFromProps;
    const oppositeColor = useThemeColor({ }, "text").colorFromProps;
    return <DefaultTouchableOpacity style={[{ backgroundColor, borderColor: oppositeColor, shadowColor: oppositeColor }, style]} containerStyle={[{ backgroundColor, borderColor: oppositeColor }, containerStyle]} {...otherProps} />
}

export const Icon = (props: IconProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "icon").colorFromProps;
    return <DefaultIcon tvParallaxProperties={undefined} color={color} {...otherProps} />
}

export const Input = (props: InputProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const { colorFromProps, theme } = useThemeColor({ light: lightColor, dark: darkColor }, "input");
    const backgroundColor = colorFromProps;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;
    return (

        <>
            <DefaultInput
                style={[{ backgroundColor: "transparent", color }, style]}
                keyboardAppearance={theme}
                {...otherProps}
            />
            <View style={{ backgroundColor, width: "100%", height: "100%", opacity: .5, position: "absolute", borderRadius: 15, zIndex: -10 }} />
        </>
    )
    // return <DefaultInput style={[{backgroundColor: "transparent", color }, style]} keyboardAppearance={theme} {...otherProps} />
}

export const ScrollView = (props: ScrollViewProps) => {
    const { style, darkColor, lightColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    return <DefaultScrollView style={[{ backgroundColor }, style]} {...otherProps} />
}

export const ButtonGroup = (props: ButtonGroupProps) => {
    const { containerStyle, darkColor, lightColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "input").colorFromProps;
    
    return <DButtonGroup containerStyle={[{ backgroundColor: color, borderColor: color }, containerStyle]}  {...otherProps} />
}