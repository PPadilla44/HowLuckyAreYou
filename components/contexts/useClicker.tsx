import React, { createContext, Dispatch, ReactNode, useContext, useReducer } from 'react';
import { Action, initialClickerState, ClickerState } from '../../store/clicker';
import clickerReducer from "../../store"

const ClickerContext = createContext<{ state: ClickerState, dispatch?: Dispatch<Action>  }>({ state: initialClickerState });

export const useClicker = () => {
    return useContext(ClickerContext);
}

type Props = {
    children: ReactNode;
};

export const ClickerProvider = ({ children }: Props) => {

    const [state, dispatch] = useReducer(clickerReducer, initialClickerState);

    const providerState = {
        state,
        dispatch
    }

    return (
        <ClickerContext.Provider value={providerState}>
            {children}
        </ClickerContext.Provider>
    )
}
