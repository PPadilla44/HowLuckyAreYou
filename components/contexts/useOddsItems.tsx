import React, { createContext, Dispatch, ReactNode, useContext, useReducer } from 'react';
import { Action,  } from '../../store/oddsItems';
import oddsItemsReducer from "../../store/oddsItems"
import { OddsItemInterface } from '../../types';

const OddsItems = createContext<{ state: OddsItemInterface[], dispatch?: Dispatch<Action>  }>({ state: [] });

export const useOddsItems = () => {
    return useContext(OddsItems);
}

type Props = {
    children: ReactNode;
};

export const OddsItemsProvider = ({ children }: Props) => {

    const [state, dispatch] = useReducer(oddsItemsReducer, [] );

    const providerState = {
        state,
        dispatch
    }

    return (
        <OddsItems.Provider value={providerState}>
            {children}
        </OddsItems.Provider>
    )
}
