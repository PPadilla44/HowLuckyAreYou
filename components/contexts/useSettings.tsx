import { createContext, Dispatch, ReactNode, useContext, useReducer } from "react";
import { Appearance } from "react-native";
import settingsReducer from "../../store/settings"
import { Action, initialSettingsState, SettingsState } from "../../store/settings";

const Settings = createContext<{ state: SettingsState , dispatch?: Dispatch<Action> }>( { state: initialSettingsState } );

export const useSettings = () => {
    return useContext(Settings);
}


type Props = {
    children: ReactNode;
};

export const SettingsProvider = ({ children }: Props) => {

    const [state, dispatch] = useReducer(settingsReducer, initialSettingsState );

    if(state.data.appearance.fromDevice) {
        state.data.appearance.appearance = Appearance.getColorScheme() as "light" | "dark";
    }

    const providerState = {
        state,
        dispatch
    }

    return (
        <Settings.Provider value={providerState}>
            {children}
        </Settings.Provider>
    )
}
