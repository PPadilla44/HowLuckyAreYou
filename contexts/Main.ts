import { useReducer } from "react";
import { View } from "react-native";
import CustomContext from "./context";

export interface MainState {
    count: number;
    odds: number;
    didHit: boolean;
    results: {
        BtnColor: string;
        text: string;
    };
}

export enum MainActionKind {
    INCREASE = "INCREASE",
}

export interface Action {
    type: MainActionKind,
    payload?: any
}

export const initialMainState = {
    count: 0,
    odds: .1,
    didHit: false,
    results: {
        BtnColor: "",
        text: ""
    }
}

function increment(state: MainState) {
    console.log("****************");

    const tempCount = state.count + 1;

    const max = 1 / (state.odds);

    const randomNumFromMax = Math.ceil(Math.random() * max);

    console.log(randomNumFromMax, max);

    if (randomNumFromMax === max) {
        console.log("GOT IT");
        const userOdds = tempCount / max;
        let tempResults
        if (userOdds < 1) {
            tempResults = { BtnColor: "green", text: "YOU ARE REALLY LUCKY" };
        } else if (userOdds > 1) {
            tempResults = { BtnColor: "red", text: "YOU ARE NOT LUCKY" };
        } else {
            tempResults = { BtnColor: "blue", text: "YOU ARE normal" };
        }
        const newState = {
            ...state,
            count: tempCount,
            results: tempResults,
            didHit: true
        }
        console.log(newState);

        return newState
    }

    return {
        ...state,
        count: tempCount
    }
}

export function mainReducer(state: MainState = initialMainState, action: Action) {
    const { type, payload } = action;
    switch (type) {
        case MainActionKind.INCREASE:
            return increment(state)

        default:
            return state;
    }
}

// const ParentComponent = ({ children }) => {

//     const [userState, usersDispatch] = useReducer(mainReducer, initialMainState);

//     const providerState = {
//         userState,
//         usersDispatch
//     }

//     return (
//         <View>
//         </View>
//     )
// }

// export default ParentComponent;