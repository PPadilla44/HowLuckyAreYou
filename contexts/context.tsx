import React, { createContext, useContext } from 'react';

const CustomContext = createContext(null);

export function useCustomContext() {
    return useContext(CustomContext);
}

export default CustomContext;