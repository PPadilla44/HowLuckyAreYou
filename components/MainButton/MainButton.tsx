import { StyleSheet } from 'react-native'
import { Text, View, TouchableOpacity } from "../../components/Themed";

import React, { FC, useReducer } from 'react'
import { mainReducer, initialMainState, MainActionKind } from '../../contexts/Main';

const MainButton = () => {

    const [state, dispatch] = useReducer(mainReducer, initialMainState);
    const { didHit, results } = state;
    

    const handlePress = () => {
        dispatch({ type: MainActionKind.INCREASE })
    }

    if (didHit) {
        return (
            <View style={[styles.btnDone, { backgroundColor: results.BtnColor }]} >
                <Text style={styles.btnDoneText}>{results.text}</Text>
            </View>
        )
    }

    return (
        <TouchableOpacity style={styles.btn} onPress={handlePress} activeOpacity={0.7}>
            <Text style={styles.btnText}>Test Your Luck</Text>
        </TouchableOpacity>
    )
}

export default MainButton

const styles = StyleSheet.create({
    btn: {
        width: 200,
        height: 200,
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 100,
        shadowColor: '#000000',
        shadowOffset: { width: 0, height: 10 },
        shadowOpacity: 0.1,
        shadowRadius: 3,
    },
    btnDone: {
        width: 200,
        height: 200,
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 100,
        shadowColor: '#000000',
        shadowOffset: { width: 0, height: 10 },
        shadowOpacity: 0.1,
        shadowRadius: 3,
    },
    btnText: {
        fontSize: 24,
        fontWeight: "bold",
        color: "white",
        fontFamily: "LexendDeca_700Bold",
        textAlign: "center"
    },
    btnDoneText: {
        fontSize: 24,
        fontWeight: "bold",
        color: "black",
        fontFamily: "LexendDeca_700Bold",
        textAlign: "center"

    },

})