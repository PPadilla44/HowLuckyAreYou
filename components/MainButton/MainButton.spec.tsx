import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import { ClickerProvider } from "../contexts/useClicker";
import { TouchableOpacity } from "../Themed";
import MainButton from "./MainButton";


describe('<MainButton', () => {

    afterEach(cleanup);

    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(
            <ClickerProvider>
                <MainButton />
            </ClickerProvider>
        )
    });

    it('should render correctly', () => {
        wrapper.getByText("What Are The Odds?");
        wrapper.container.findByType(TouchableOpacity)
    });

    it('should should be able to press button until results view mounted', () => {
        const btn = wrapper.getByTestId("mainBtn");
        while(wrapper.container.findAllByType(TouchableOpacity).length > 0) {
            fireEvent.press(btn);
        };
        wrapper.getByTestId("mainBtnResults");
    });

})
