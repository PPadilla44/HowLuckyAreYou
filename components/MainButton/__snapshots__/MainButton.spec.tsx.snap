// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainButton should match snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Array [
      Object {
        "backgroundColor": "#325C99",
        "borderColor": "white",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "height": 200,
        "justifyContent": "center",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
        "width": 200,
      },
    ]
  }
  testID="mainBtn"
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "#325C99",
        "borderColor": "white",
        "opacity": 1,
        "shadowColor": "white",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Futura",
          },
          Object {
            "color": "white",
            "fontFamily": "Futura",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Test Your Luck
    </Text>
  </View>
</RNGestureHandlerButton>
`;
